### PR TITLE
test: httpCoreClient

### PR DESCRIPTION
**Pull request by Fun795**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -73,6 +73,8 @@
     "@byndyusoft/nest-dynamic-module": "^1.0.0",
     "deepmerge": "^4.2.2",
     "lodash": "^4.17.21",
+    "reflect-metadata": "^0.1.13",
+    "rxjs": "^7.5.6",
```
Comment (KillWolfVlad) [inline package.json:77]: Please, remove it from dependencies

File: src/__tests__/httpCoreClient.spec.ts [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+ * limitations under the License.
+ */
+
+import { default as axios } from "axios";
+
+import { HttpCoreClient } from "../httpCoreClient";
+
+jest.mock("axios");
+
+const axiosMock = axios as jest.Mocked<typeof axios>;
```
Comment (KillWolfVlad) [inline src/__tests__/httpCoreClient.spec.ts:23]: Mock axios like this:

```typescript
import { AxiosInstance } from "axios";
import { mock, MockProxy } from "jest-mock-extended";

import { HttpCoreClient } from "../httpCoreClient";

describe("httpCoreClient", () => {
  let axios: MockProxy<AxiosInstance>;

  let httpCoreClient: HttpCoreClient;

  beforeAll(() => {
    axios = mock();

    httpCoreClient = new HttpCoreClient({
      axios,
      config: {
        baseURL: "localhost",
      },
    });
  });
```

**Review by KillWolfVlad — CHANGES_REQUESTED** [review]
- [ ] Add tests for `HttpClient`
- [ ] Use `it.each` for more readability
- [ ] ~~Add tests for headers merge issue (moved to outer issue https://github.com/Byndyusoft/nest-http-client/issues/44)~~

File: package.json (hunk 1 above)
Comment (Fun795) [reply package.json]: moved to devDependencies

File: package.json [hunk 3]
```diff
@@ -72,6 +72,7 @@
   "dependencies": {
     "@byndyusoft/nest-dynamic-module": "^1.0.0",
     "deepmerge": "^4.2.2",
+    "jest-mock-extended": "^3.0.1",
```
Comment (KillWolfVlad) [inline package.json:75]: Move to dev deps

**Review by KillWolfVlad — CHANGES_REQUESTED** [review]

**Comment by KillWolfVlad** [comment]
Closed in favor of #46

**Pull request closed without merging**